### PR TITLE
Fix new message navigation bar style

### PR DESCRIPTION
**Pull request by Attila-Varga-Instructure**
refs: MBL-10467
affects: Teacher
release note: Apply color theme to new message dialog navigation bar.

test plan:
- Enter Inbox.
- Start compose a new message.
- Nav bar should reflect the account's theme.
- Enter Select a Course screen.
- Nav bar should reflect the account's theme.
- Select course.
- Tap + sign on the To field.
- Nav bar should reflect the account's theme.

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Yarn lint</summary>


    
    /Users/vagrant/git/rn/Teacher/src/modules/inbox/__tests__/CourseSelect.test.js
      90:1  error  Expected indentation of 4 spaces but found 6  indent
      91:1  error  Expected indentation of 4 spaces but found 6  indent
    
    ✖ 2 problems (2 errors, 0 warnings)
      2 errors and 0 warnings potentially fixable with the `--fix` option.
    
</details>


[MBL-10467](https://instructure.atlassian.net/browse/MBL-10467)

#### Affected Apps: Teacher

#### Release Note: 
Apply color theme to new message dialog navigation bar.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53018438/p/73638489941454519e506c4540b2cf90">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53018469/p/b88a717f0256ab6a3806df401bf1359a">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53018406/p/ff7aabb828e37fbda79f255d052e8941">
    
  </a>
</details>

File: rn/Teacher/src/modules/address-book/AddressBook.js [hunk 1]
```diff
@@ -197,6 +197,7 @@ export class AddressBook extends Component<AddressBookProps, State> {
     return (
       <Screen
         drawUnderNavBar
+        navBarStyle='global'
```
Comment (vanwagonet) [inline rn/Teacher/src/modules/address-book/AddressBook.js:200]: I think the style we want for these is `'modal'`. The bar background will end up white, with the buttons tinted the linkColor.

File: rn/Teacher/src/modules/inbox/Compose.js [hunk 2]
```diff
@@ -236,6 +236,7 @@ export class Compose extends PureComponent<ComposeProps & OwnProps, ComposeState
     return (
       <Screen
         drawUnderNavBar
+        navBarStyle='global'
```
Comment (vanwagonet) [inline rn/Teacher/src/modules/inbox/Compose.js:239]: modal

File: rn/Teacher/src/modules/inbox/CourseSelect.js [hunk 3]
```diff
@@ -84,6 +84,7 @@ export class CourseSelect extends PureComponent<CourseSelectProps> {
       <Screen
         title={i18n('Select a Course')}
         drawUnderNavBar
+        navBarStyle='context'
```
Comment (vanwagonet) [inline rn/Teacher/src/modules/inbox/CourseSelect.js:87]: modal

**Review by vanwagonet — CHANGES_REQUESTED** [review]

File: rn/Teacher/src/modules/address-book/AddressBook.js (hunk 1 above)
Comment (Attila-Varga-Instructure) [reply rn/Teacher/src/modules/address-book/AddressBook.js:200]: Modal just displayed the standard iOS style. Is this a bug? 
Simulator Screen Shot - iPhone 11 - 2020-10-19 at 15 59 04

Comment (vanwagonet) [reply rn/Teacher/src/modules/address-book/AddressBook.js:200]: That looks correct, assuming the brand for your sandbox has electric blue for the linkColor.

Comment (Attila-Varga-Instructure) [reply rn/Teacher/src/modules/address-book/AddressBook.js]: Yes, that was the setting, thanks. I forgot these are used in the app as well. I've modified the PR accordingly.

**Review by vanwagonet — DISMISSED** [review]
QA+1

**Review by vanwagonet — APPROVED** [review]

**Pull request merged**